bindings/pyroot/cppyy/CPyCppyy/src/CPPConstructor.cxx

```diff
@@ -83,10 +84,9 @@ PyObject* CPyCppyy::CPPConstructor::Call(
         address = (ptrdiff_t)((CPPInstance*)pyobj)->GetObject();
         if (address) {
             ((CPPInstance*)pyobj)->CppOwns();
-            PyObject* pyoff = PyObject_CallMethod(dispproxy, (char*)"_dispatchptr_offset", nullptr);
-            size_t disp_offset = PyLong_AsSsize_t(pyoff);
-            Py_DECREF(pyoff);
-            new ((void*)(address + disp_offset)) DispatchPtr{(PyObject*)self};
+            PyObject* res = PyObject_CallMethodObjArgs(
+            dispproxy, PyStrings::gDispInit, pyobj, (PyObject*)self, nullptr);
+            Py_XDECREF(res);
```

Comment:
Just for my understanding, there's a reason you used the `XDECREF` and not a simple `DECREF`?

Reply:
I'd say the reason is that `PyObject_CallMethodObjArgs` can return null in case of failure, and `XDECREF` handles null unlike `DECREF`.